app/controllers/reports/regions_controller.rb

```diff
@@ -101,10 +101,21 @@ def details
 
     @repository = Reports::Repository.new(regions, periods: @period_range, follow_ups_v2: current_admin.feature_enabled?(:follow_ups_v2))
     chart_repo = Reports::Repository.new(@region, periods: chart_range, follow_ups_v2: current_admin.feature_enabled?(:follow_ups_v2))
+    medications_dispensation_service = MedicationDispensationService.new(region: @region, period: @period)
+
+    district_regions = if @region.state_region?
+      [@region, @region.district_regions].flatten
+    end
+
+    if @region.state_region?
+      @district_repository = Reports::Repository.new(district_regions, periods: @period_range, follow_ups_v2: current_admin.feature_enabled?(:follow_ups_v2))
+    end
 
     @chart_data = {
       patient_breakdown: PatientBreakdownService.call(region: @region, period: @period),
-      ltfu_trend: ltfu_chart_data(chart_repo, chart_range)
+      ltfu_trend: ltfu_chart_data(chart_repo, chart_range),
+      medications_dispensation: medications_dispensation_service.distribution,
+      medications_dispensation_months: medications_dispensation_service.months
```

Comment:
Alternative 1
```
medications_dispensation: {distribution: medications_dispensation_service.distribution,
                                 months: medications_dispensation_service.months}
```

Alternative 2
Drop the months list and build the legends by parsing the distribution data structure in JS